### ISSUE TITLE
[ERROR] Cannot install the requirements on Linux

### ISSUE THREAD

**Issue by wutipong** [victorchall/EveryDream2trainer#239]
**To Reproduce**
Steps to reproduce the undesired behavior:
1.  Clone the project on Github
2.  Create venv using Python 3.10
3. Activate venv
4.  Run `pip install -r requirements.txt`

**Attach log and cfg**

```
INFO: pip is looking at multiple versions of xformers to determine which version is compatible with other requirements. This could take a while.
ERROR: Cannot install -r requirements.txt (line 12), -r requirements.txt (line 2), diffusers[torch]==0.21.4 and torch==2.1.0 because these package versions have conflicting dependencies.

The conflict is caused by:
    The user requested torch==2.1.0
    torchvision 0.16.0 depends on torch==2.1.0
    diffusers[torch] 0.21.4 depends on torch>=1.4; extra == "torch"
    xformers 0.0.20 depends on torch==2.0.1

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
```

**Runtime environment (please complete the following information):**
 - OS: Arch Linux, Kernel 5.15, Nvidia Driver 535.129.03
 - Python 3.10.13
 - PIP 23.2.1
 - Local PC (Core i7 4790, 16GB Memory)
 - GPU RTX3060 12GB

**Additional context**

Also consider posting your cfg and log to the Discord #help channel here instead, there are far more people there than will read your issue here on Github: 
Have you joined discord?  You're far more likely to get a response there: https://discord.gg/uheqxU6sXN

**Closure**
Reminder:  Please close your issue once it has been resolved and please state what you did to fix it, whether a reply helped you or you figure it out on your own.  Your issue may be closed if you do not reply to keep the backlog of issues manageable.

**Comment by victorchall** [comment]
requirements has been updated to use xformers 0.0.22.post7
That should take care of it.  Reopen if it does not.

**Comment by aligg73** [comment]
Changing to `xformers==0.0.22.post7e` alone unfortunately didn't fix it for me.

On Ubuntu, the only way I could get it to work was to piggy-back what's happening in the Dockerfile:

`pip install -U -I torch==2.1.0+cu121 torchvision==0.16.0+cu121 --extra-index-url "https://download.pytorch.org/whl/cu121" && \
    pip install -r requirements-build.txt && \
    pip install --no-deps xformers==0.0.22.post7`
    
and:
    
    `pip install --no-cache-dir -r requirements-runtime.txt`

Not sure if it leads to any issues downstream but at least I could get past the install step.
Hope that helps, cheers.

**Comment by dvaupel** [comment]
A fresh install still leads to the following error message:
```
ERROR: Cannot install -r requirements.txt (line 12), -r requirements.txt (line 13), -r requirements.txt (line 2), -r requirements.txt (line 24), diffusers[torch]==0.21.4 and torch==2.1.0 because these package versions have conflicting dependencies.

The conflict is caused by:
    The user requested torch==2.1.0
    torchvision 0.16.0 depends on torch==2.1.0
    diffusers[torch] 0.21.4 depends on torch>=1.4; extra == "torch"
    xformers 0.0.22.post7 depends on torch==2.1.0
    pytorch-lightning 1.6.5 depends on torch>=1.8.*
    open-flamingo 2.0.0 depends on torch==2.0.1

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
```

If you remove the package versions from torch, torchvision and open-flamingo it seems to work.

**Comment by victorchall** [comment]
Open flamingo does not support torch 2.1.  If you install open flamingo it will back date to torch 2.0.1.  There's nothing I can do about that until open flamingo updates.  I tried `pip install --no-deps openflamingo ` to see if I could force it to work with torch 2.1 and it fails to run. 

I've moved the project on to torch 2.1, breaking open flamingo, because torch 2.1 has distributions available for cuda 12.1 which is a massive performance increase for Ada Lovelace GPUs without having to try to do weird stuff with cuda DLL files. 

Open flamingo is not critical to the project, and at this point with more Ada GPUs making their way into people's hands Torch 2.1 IS important and I don't feel it is practical to support multiple torch versions from my perspective.   

I will just remove open flamingo from requirements. If you want to use open flamingo you can setup a separate venv or entirely separate clone, then you can run requirements.txt and add then pip install open flamingo.  

If this becomes a major problem for people my choice would simply be to remove open flamingo entirely from this repo as it is not worth trying to deal with the issues. If someone else wants to make specific open flamingo repo just for open flamingo that might be a better idea, but it's not something I will have time to support.

**Comment by victorchall** [comment]
I've updated `requirements.txt` to remove open flamingo and added documentation on the problems in the CAPTION.md file to outline why.

All I can suggest is you create a separate environment just for open flamingo.  You can use a  separate venv if you want and are  familiar with doing that and you can swap which venv you are activating to use open flamingo, or just make an entirely separate clone of this repo, run through the normal install, then `pip install open-flamingo` and only use that specific copy/clone for caption_fl.py only. 

You'll have to ask ML Foundations if they want to update their package to support torch 2.1.
